src/transport/SecureSessionTable.h

```diff
@@ -147,6 +147,31 @@ class SecureSessionTable
         return result != nullptr ? MakeOptional<SessionHandle>(*result) : Optional<SessionHandle>::Missing();
     }
 
+    // Select SessionHolders which are pointing a session with the same peer as the given session. Shift them the the given session.
+    // This is an internal API, using raw pointer to a session is allowed here.
+    void NewerSessionAvailable(SecureSession * session)
+    {
+        VerifyOrDie(session->GetSecureSessionType() == SecureSession::Type::kCASE);
+        mEntries.ForEachActiveObject([&](SecureSession * oldSession) {
+            if (session == oldSession)
+                return Loop::Continue;
+
+            SessionHandle ref(*oldSession);
+
+            // This will update all SessionHolder pointing to oldSession, to the provided session.
```

Comment:
No, it won't.  Because some of them revise to be updated, like the second half of the comment says.  How about:
```suggestion
            // This will give all SessionHolders pointing to oldSession a chance to switch to the provided session.
```